src/main/java/org/justinbaur/bankteller/service/UserProfileServiceDatabaseImpl.java

```diff
@@ -0,0 +1,63 @@
+package org.justinbaur.bankteller.service;
+
+import org.justinbaur.bankteller.domain.Profile;
+import org.justinbaur.bankteller.exceptions.AccountNotFound;
+import org.justinbaur.bankteller.exceptions.ProfileNotFound;
+import org.justinbaur.bankteller.exceptions.UpdateException;
+import org.justinbaur.bankteller.repository.ProfileRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserProfileServiceDatabaseImpl extends ProfileServiceImpl implements UserProfileService {
+
+    @Autowired
+    ProfileRepository repository;
+
+    @Override
+    public Integer getBalance(String profileId, String accountName) throws ProfileNotFound, AccountNotFound {
+        Profile profile = getProfile(profileId);
+        if (profile != null) {
```

Comment:
This would never be null as your `getProfile` method will return an exception, so there is no need to check it here.